lib/ex_json_path.ex

```diff
@@ -17,6 +17,31 @@
 #
 
 defmodule ExJsonPath do
+  @moduledoc """
+  This module implements a JSONPath evaluator.
+  """
+
+  @type compiled_path :: list({:access, String.t()})
```

Comment:
I'd use `@opaque` instead of `@type` since the internal representation of a compiled path is an implementation detail that you don't want to expose